batcher/aligned-sdk/src/core/types.rs

```diff
@@ -196,32 +255,18 @@ impl ClientMessage {
     /// The signature of the message is verified, and when it correct, the
     /// recovered address from the signature is returned.
     pub fn verify_signature(&self) -> Result<Address, SignatureError> {
-        let hashed_data: [u8; 32] =
-            ClientMessage::hash_with_nonce_and_chain_id(&self.verification_data);
+        let recovered = self.signature.recover_typed_data(&self.verification_data)?;
+
+        // We can expect here because encode_eip712 can only error if
+        // struct_hash or domain_separator return an error, which is not possible
+        let hashed_data = self
+            .verification_data
+            .encode_eip712()
+            .expect("Failed to encode verification data for signature verification");
```

Comment:
👮 return error instead of expect

Reply:
I used expect because of this:
```
We can expect here because encode_eip712 can only error if
struct_hash or domain_separator return an error, which is not possible
```
I added it as a comment in lines 260 and 261

Reply:
I used expect because `encode_eip712` couldn't error since we defined `struct_hash` and `domain_separator` as I mentioned in the comment above the function call. But you are right it is better to return the error anyways. I solved it in [e8be0d7](https://github.com/yetanotherco/aligned_layer/pull/916/commits/e8be0d7b553b1fb80c0f96a60fa491a116a47526) adding a custom error.